src/EnergyPlus/UserDefinedComponents.cc

```diff
@@ -131,6 +131,8 @@ namespace UserDefinedComponents {
                 EMSManager::ManageEMS(state, EMSManager::EMSCallFrom::UserDefinedComponentModel, anyEMSRan, this->Loop(thisLoop).ErlInitProgramMngr);
             } else if (this->Loop(thisLoop).initPluginLocation > -1) {
                 state.dataPluginManager->pluginManager->runSingleUserDefinedPlugin(state, this->Loop(thisLoop).initPluginLocation);
+            } else if (this->Loop(thisLoop).initCallbackIndex > -1) {
+                state.dataPluginManager->pluginManager->runSingleUserDefinedCallback(state, this->Loop(thisLoop).initCallbackIndex);
```

Comment:
This is really the only change in the user defined code: we try to get an index to a callback if it can't find that program name as an EMS program or Python Plugin, and then later we call the callback if it's found.  That's it.